### PR TITLE
Add RunAsUser support

### PR DESCRIPTION
**Pull request by Random-Liu**
This PR:
1) Update containerd to include the new `WithUsername`.
2) Fix several security related issues, including:
  a) Set NoNewPrivs no matter the container is privileged or not (same with [what docker does](https://github.com/moby/moby/blob/master/daemon/oci_linux.go#L832)) https://github.com/kubernetes-incubator/cri-containerd/issues/166#issuecomment-324725784.
  b) Set RunAsUser and SupplementalGroups for sandbox container.
  c) Return error for privileged container if its corresponding sandbox doesn't allow privilege.
3) Add RunAsUsername support leveraging https://github.com/containerd/containerd/pull/1413.

/cc @zhangzhenhao

### CODE REVIEW

File: hack/test-cri.sh [hunk 1]
```diff
@@ -21,7 +21,7 @@ source $(dirname "${BASH_SOURCE[0]}")/test-utils.sh
 # FOCUS focuses the test to run.
 FOCUS=${FOCUS:-}
 # SKIP skips the test to skip.
-SKIP=${SKIP:-"RunAsUserName"}
```
Comment (Random-Liu) [inline hack/test-cri.sh]: No default skip anymore! Yay!

File: pkg/server/container_create.go [hunk 2]
```diff
@@ -135,8 +135,16 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 		containerMetadataLabel: string(metaBytes),
 	}
 
+	specOpts := containerd.WithSpec(spec)
+	// Set container username. This could only be done by containerd, because it needs
+	// access to the container rootfs. Pass user name to containerd, and let it overwrite
+	// the spec for us.
+	if username := config.GetLinux().GetSecurityContext().GetRunAsUsername(); username != "" {
+		specOpts = containerd.WithSpec(spec, containerd.WithUsername(username))
```
Comment (mikebrow) [inline pkg/server/container_create.go]: looks right...

File: pkg/server/container_create.go [hunk 3]
```diff
@@ -233,14 +244,15 @@ func (c *criContainerdService) generateContainerSpec(id string, sandboxPid uint3
...
 		// TODO(random-liu): [P2] Add apparmor and seccomp.
 
-		// TODO: Figure out whether we should set no new privilege for sandbox container by default
-		g.SetProcessNoNewPrivileges(securityContext.GetNoNewPrivs())
 	}
 
+	// TODO: Figure out whether we should set no new privilege for sandbox container by default
+	g.SetProcessNoNewPrivileges(securityContext.GetNoNewPrivs())
+
+	// TODO(random-liu): [P1] Set selinux options.
```
Comment (mikebrow) [inline pkg/server/container_create.go]: // TODO(mikebrow): [P1] Set apparmor and seccomp options

File: vendor.conf [hunk 4]
```diff
@@ -1,6 +1,6 @@
 github.com/blang/semver v3.1.0
 github.com/boltdb/bolt v1.3.0-58-ge9cf4fa
-github.com/containerd/containerd f05281743e5ac9ad11c6e19a72be7a903eab79f5 
+github.com/containerd/containerd a6ce1ef2a140d79856a8647e1d1ae5ac9ab581eb
```
Comment (mikebrow) [inline vendor.conf]: I verified your vendor update, looks right.

**Pull request merged**